### DISCUSSION TITLE
Apache container not running

### DISCUSSION THREAD

**Discussion by tfedder-de** [nextcloud/all-in-one#877] [Questions] [answered]
Hi there,

tl;dr
- Nextcloud AIO v1.4.2
- only Apache container not running
- using Caddy as reverse proxy like recommended

I am trying to setup Nextcloud AIO v1.4.2. After some iterations of trail and error, including stopping and removing containers and volumes, I am stuck with the Apache Container not running.

I have a Caddy as a reverse proxy installed on the same host that runs the docker containers, anologous to the recommendation in https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md and basically followed the example for docker-compose that is linked in there.

I added the Caddy because it seemed to me that I was not permitted to use ports 80 and 443 via docker for some reason. Since I am planning to host additional web services via Caddy from that machine anyway, I just went with it.

Nonetheless, after starting the containers via the Nextcloud AIO WebUI - and most of them turning green - the Apache container eventually turned red (Stopped).

Sending a request to the domain returns a HTTP 502, probably because the Apache isn't running. I thought for a moment #594 might be related but, at least to me, it seems to be a different error.

Any help would be appreciated.

<details><summary>Apache Log via Nextcloud AIO WebUI</summary>
nc: getaddrinfo for host "nextcloud-aio-nextcloud" port 9000: Name or service not known
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
runtime: failed to create new OS thread (have 4 already; errno=11)
runtime: may need to increase max user processes (ulimit -u)
fatal error: newosproc

runtime stack:
runtime.throw({0x194ed93?, 0x7ffc77321af0?})
	runtime/panic.go:992 +0x71
runtime.newosproc(0xc000060800)
	runtime/os_linux.go:182 +0x185
runtime.newm1(0xc000060800)
	runtime/proc.go:2147 +0xcf
runtime.newm(0x27e0a40?, 0xc00004c500, 0x443633?)
	runtime/proc.go:2122 +0x12c
runtime.startm(0x0, 0x1)
	runtime/proc.go:2305 +0xcf
runtime.wakep()
	runtime/proc.go:2404 +0x5a
runtime.newproc.func1()
	runtime/proc.go:4062 +0x53
runtime.systemstack()
	runtime/asm_amd64.s:469 +0x49

goroutine 1 [running, locked to thread]:
runtime.systemstack_switch()
	runtime/asm_amd64.s:436 fp=0xc00005c600 sp=0xc00005c5f8 pc=0x469940
runtime.newproc(0x18000000000?)
	runtime/proc.go:4055 +0x51 fp=0xc00005c638 sp=0xc00005c600 pc=0x443351
runtime.init.6()
	runtime/proc.go:289 +0x25 fp=0xc00005c650 sp=0xc00005c638 pc=0x43afc5
runtime.doInit(0x27cd680)
	runtime/proc.go:6222 +0x126 fp=0xc00005c780 sp=0xc00005c650 pc=0x447e46
runtime.main()
	runtime/proc.go:199 +0x108 fp=0xc00005c7e0 sp=0xc00005c780 pc=0x43ad28
runtime.goexit()
	runtime/asm_amd64.s:1571 +0x1 fp=0xc00005c7e8 sp=0xc00005c7e0 pc=0x46ba21
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.8. Set the 'ServerName' directive globally to suppress this message
runtime: failed to create new OS thread (have 2 already; errno=11)
runtime: may need to increase max user processes (ulimit -u)
fatal error: newosproc

runtime stack:
runtime.throw({0x194ed93?, 0x7fffd5abe858?})
	runtime/panic.go:992 +0x71
runtime.newosproc(0xc000060000)
	runtime/os_linux.go:182 +0x185
runtime.newm1(0xc000060000)
	runtime/proc.go:2147 +0xcf
runtime.newm(0x443351?, 0x0, 0x287ec20?)
	runtime/proc.go:2122 +0x12c
runtime.main.func1()
	runtime/proc.go:171 +0x29
runtime.systemstack()
	runtime/asm_amd64.s:469 +0x49

goroutine 1 [running]:
runtime.systemstack_switch()
	runtime/asm_amd64.s:436 fp=0xc00005c780 sp=0xc00005c778 pc=0x469940
runtime.main()
	runtime/proc.go:170 +0x6d fp=0xc00005c7e0 sp=0xc00005c780 pc=0x43ac8d
runtime.goexit()
	runtime/asm_amd64.s:1571 +0x1 fp=0xc00005c7e8 sp=0xc00005c7e0 pc=0x46ba21
runtime: failed to create new OS thread (have 6 already; errno=11)
runtime: may need to increase max user processes (ulimit -u)
fatal error: newosproc

runtime stack:
runtime.throw({0x194ed93?, 0xc000071e28?})
	runtime/panic.go:992 +0x71
runtime.newosproc(0xc000180000)
	runtime/os_linux.go:182 +0x185
runtime.newm1(0xc000180000)
	runtime/proc.go:2147 +0xcf
runtime.newm(0xc000071f08?, 0xc00004c500, 0x1?)
	runtime/proc.go:2122 +0x12c
runtime.startm(0x0, 0x1)
	runtime/proc.go:2305 +0xcf
runtime.wakep()
	runtime/proc.go:2404 +0x5a
runtime.resetspinning()
	runtime/proc.go:3036 +0x45
runtime.schedule()
	runtime/proc.go:3194 +0x25e
runtime.park_m(0xc000002680?)
	runtime/proc.go:3336 +0x14d
runtime.mcall()
	runtime/asm_amd64.s:425 +0x43

goroutine 1 [chan receive, locked to thread]:
runtime.gopark(0x7f36719e8108?, 0xc00005c720?, 0xc5?, 0xd6?, 0x0?)
	runtime/proc.go:361 +0xd6
runtime.chanrecv(0xc00007c000, 0x0, 0x1)
	runtime/chan.go:577 +0x56c
runtime.chanrecv1(0x0?, 0x2?)
	runtime/chan.go:440 +0x18
runtime.gcenable()
	runtime/mgc.go:179 +0xb6
runtime.main()
	runtime/proc.go:209 +0x13d
runtime.goexit()
	runtime/asm_amd64.s:1571 +0x1
{"level":"info","ts":1656533456.0913801,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
runtime: failed to create new OS thread (have 8 already; errno=11)
runtime: may need to increase max user processes (ulimit -u)
fatal error: newosproc

goroutine 53 [running]:
runtime.throw({0x194ed93?, 0xc0004d3e70?})
	runtime/panic.go:992 +0x71 fp=0xc0004d3e38 sp=0xc0004d3e08 pc=0x438711
runtime.newosproc(0xc000061400)
	runtime/os_linux.go:182 +0x185 fp=0xc0004d3e98 sp=0xc0004d3e38 pc=0x434ca5
runtime.newm1(0xc000061400)
	runtime/proc.go:2147 +0xcf fp=0xc0004d3ed8 sp=0xc0004d3e98 pc=0x43eccf
runtime.newm(0x0?, 0x0, 0x0?)
	runtime/proc.go:2122 +0x12c fp=0xc0004d3f08 sp=0xc0004d3ed8 pc=0x43eb8c
runtime.startTemplateThread()
	runtime/proc.go:2167 +0x90 fp=0xc0004d3f38 sp=0xc0004d3f08 pc=0x43edb0
runtime.LockOSThread()
	runtime/proc.go:4339 +0x1d fp=0xc0004d3f58 sp=0xc0004d3f38 pc=0x443fdd
runtime.ensureSigM.func1()
	runtime/signal_unix.go:959 +0x2e fp=0xc0004d3fe0 sp=0xc0004d3f58 pc=0x44eeee
runtime.goexit()
	runtime/asm_amd64.s:1571 +0x1 fp=0xc0004d3fe8 sp=0xc0004d3fe0 pc=0x46ba21
created by runtime.ensureSigM
	runtime/signal_unix.go:956 +0xbd

goroutine 1 [runnable]:
github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile.parseOptStorage(0xc0003f0360, {0xc0004a20c0?, 0xc0001a7a20?})
	github.com/caddyserver/caddy/v2@v2.5.1/caddyconfig/httpcaddyfile/options.go:179 +0xc9
github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile.ServerType.evaluateGlobalOptionsBlock({}, {0xc000273290?, 0x2, 0x2}, 0xd?)
	github.com/caddyserver/caddy/v2@v2.5.1/caddyconfig/httpcaddyfile/httptype.go:341 +0x4ad
github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile.ServerType.Setup({}, {0xc0003a38f0, 0x2, 0xc000287f20?}, 0xc000287e90)
	github.com/caddyserver/caddy/v2@v2.5.1/caddyconfig/httpcaddyfile/httptype.go:86 +0x11c
github.com/caddyserver/caddy/v2/caddyconfig/caddyfile.Adapter.Adapt({{0x1c9a340?, 0x28b0290?}}, {0xc000381500, 0x685, 0x686}, 0xc0003f3e60?)
	github.com/caddyserver/caddy/v2@v2.5.1/caddyconfig/caddyfile/adapter.go:50 +0x132
github.com/caddyserver/caddy/v2/cmd.LoadConfig({0x7ffebe9afe44, 0xa}, {0x0, 0x0})
	github.com/caddyserver/caddy/v2@v2.5.1/cmd/main.go:176 +0x6b2
github.com/caddyserver/caddy/v2/cmd.cmdRun({0xc000126660?})
	github.com/caddyserver/caddy/v2@v2.5.1/cmd/commandfuncs.go:206 +0x72a
github.com/caddyserver/caddy/v2/cmd.Main()
	github.com/caddyserver/caddy/v2@v2.5.1/cmd/main.go:84 +0x40d
main.main()
	./main.go:37 +0x17

goroutine 20 [select]:
github.com/caddyserver/certmagic.(*RingBufferRateLimiter).permit(0xc0003f8140)
	github.com/caddyserver/certmagic@v0.16.1/ratelimiter.go:216 +0x8c
github.com/caddyserver/certmagic.(*RingBufferRateLimiter).loop(0xc0003f8140)
	github.com/caddyserver/certmagic@v0.16.1/ratelimiter.go:89 +0x91
created by github.com/caddyserver/certmagic.NewRateLimiter
	github.com/caddyserver/certmagic@v0.16.1/ratelimiter.go:45 +0x165

goroutine 7 [chan send]:
os/signal.signal_enable(0x1?)
	runtime/sigqueue.go:220 +0x79
os/signal.enableSignal(...)
	os/signal/signal_unix.go:49
os/signal.Notify.func1(0x1724120?)
	os/signal/signal.go:145 +0x73
os/signal.Notify(0xc0002b4480, {0xc0004dcf40, 0x1, 0x0?})
	os/signal/signal.go:165 +0x1ff
github.com/caddyserver/caddy/v2.trapSignalsCrossPlatform.func1()
	github.com/caddyserver/caddy/v2@v2.5.1/sigtrap.go:40 +0x6c
created by github.com/caddyserver/caddy/v2.trapSignalsCrossPlatform
	github.com/caddyserver/caddy/v2@v2.5.1/sigtrap.go:38 +0x25

goroutine 8 [semacquire]:
sync.runtime_SemacquireMutex(0xc000058d70?, 0xc5?, 0x7fa962ad65b8?)
	runtime/sema.go:71 +0x25
sync.(*Mutex).lockSlow(0x287ef80)
	sync/mutex.go:162 +0x165
sync.(*Mutex).Lock(...)
	sync/mutex.go:81
os/signal.Notify(0xc000126180, {0xc000058f80, 0x5, 0x0?})
	os/signal/signal.go:126 +0x7a
github.com/caddyserver/caddy/v2.trapSignalsPosix.func1()
	github.com/caddyserver/caddy/v2@v2.5.1/sigtrap_posix.go:34 +0xed
created by github.com/caddyserver/caddy/v2.trapSignalsPosix
	github.com/caddyserver/caddy/v2@v2.5.1/sigtrap_posix.go:32 +0x25
runtime: failed to create new OS thread (have 8 already; errno=11)
runtime: may need to increase max user processes (ulimit -u)
fatal error: newosproc

runtime stack:
runtime.throw({0x194ed93?, 0xc00006be48?})
	runtime/panic.go:992 +0x71
runtime.newosproc(0xc0002c0400)
	runtime/os_linux.go:182 +0x185
runtime.newm1(0xc0002c0400)
	runtime/proc.go:2147 +0xcf
runtime.newm(0x427f11?, 0xc000050f00, 0x53de207b9a3?)
	runtime/proc.go:2122 +0x12c
runtime.startm(0x0, 0x1)
	runtime/proc.go:2305 +0xcf
runtime.wakep()
	runtime/proc.go:2404 +0x5a
runtime.schedule()
	runtime/proc.go:3217 +0x365
runtime.park_m(0xc00015f380?)
	runtime/proc.go:3336 +0x14d
runtime.mcall()
	runtime/asm_amd64.s:425 +0x43

goroutine 1 [runnable, locked to thread]:
github.com/yuin/goldmark/util.init()
	github.com/yuin/goldmark@v1.4.8/util/unicode_case_folding.go:3 +0x81565

goroutine 34 [select]:
github.com/caddyserver/certmagic.(*RingBufferRateLimiter).permit(0xc00022c410)
	github.com/caddyserver/certmagic@v0.16.1/ratelimiter.go:216 +0x8c
github.com/caddyserver/certmagic.(*RingBufferRateLimiter).loop(0xc00022c410)
	github.com/caddyserver/certmagic@v0.16.1/ratelimiter.go:89 +0x91
created by github.com/caddyserver/certmagic.NewRateLimiter
	github.com/caddyserver/certmagic@v0.16.1/ratelimiter.go:45 +0x165
{"level":"info","ts":1656533461.1707177,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
{"level":"warn","ts":1656533461.1733255,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/Caddyfile","line":2}
{"level":"info","ts":1656533461.1753366,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
{"level":"info","ts":1656533461.1763153,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc00043ea10"}
{"level":"warn","ts":1656533461.1772108,"logger":"http","msg":"automatic HTTPS is completely disabled for server","server_name":"srv0"}
{"level":"info","ts":1656533461.1787055,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/mnt/data/caddy"}
{"level":"info","ts":1656533461.1787465,"logger":"tls","msg":"finished cleaning storage units"}
{"level":"info","ts":1656533461.1792915,"msg":"autosaved config (load with --resume flag)","file":"/var/www/.config/caddy/autosave.json"}
{"level":"info","ts":1656533461.1793027,"msg":"serving initial configuration"}
{"level":"error","ts":1656533473.7943625,"logger":"http.log.error","msg":"dial tcp 127.0.0.1:8000: connect: connection refused","request":{"remote_ip":"172.17.0.1","remote_port":"45390","proto":"HTTP/1.1","method":"GET","host":"my.actual.domain","uri":"/status.php","headers":{"Accept-Language":["de-DE,en,*"],"X-Forwarded-For":["XX.XX.XX.XX"],"X-Forwarded-Proto":["https"],"Accept":["*/*"],"User-Agent":["Mozilla/5.0 (Windows) mirall/3.4.1stable-Win64 (build 20211221) (Nextcloud, windows-10.0.19044 ClientArchitecture: x86_64 OsArchitecture: x86_64)"],"Accept-Encoding":["gzip, deflate"],"Authorization":[],"Cookie":[],"X-Forwarded-Host":["my.actual.domain"],"X-Request-Id":["068c4b6b-cf28-4124-8c0c-077c2f7c5c66"]}},"duration":0.000847378,"status":502,"err_id":"yc34im5hy","err_trace":"reverseproxy.statusError (reverseproxy.go:1196)"}
</details>

<details><summary>Caddy config</summary>
https://my.actual.domain:443 {
    reverse_proxy http://localhost:11000
}
</details>

<details><summary>docker-compose.yml</summary>
version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer

services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    container_name: nextcloud-aio-mastercontainer
    restart: always
    ports:
      - 8080:8080
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - APACHE_PORT=11000
</details>

**Comment by szaimen** [comment]
Hello, I fear I have never seen this problem so no clue what you could try to resolve it.

Resetting the instance properly (also removing the images) and trying again is what you probably already tried by following https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance, I suppose?

**Reply by tfedder-de** [reply]
Yes, I believe I did all that.
So I guess I am trying that again soon, just to be sure.

**Reply by tfedder-de** [reply]
After a few more tries, including resetting the instance as described, even restarting the host machine that was slowing down for unknown reasons, and small changes to the Caddy config, I finally managed to get the Apache container running (and therefor turning green in the WebUI), so it doesn't get stopped any longer.

Unfortunately the problem of HTTP 502 Bad Gateway persists.

It was not, as I thought, the Caddy on the host, but the Caddy in the Apache container 🤔, that was and still is returning the HTTP 502. As it is the same when requested from the outside through the Caddy or from the host machine via `curl localhost:11000`.

So far I wasn't able to figure out what Caddy, or therefore Go, is trying to tell me with `"reverseproxy.statusError (reverseproxy.go:1196)"`

<details><summary>Excerpts from Log</summary><code>
{"remote_ip":"172.17.0.1","remote_port":"49540","proto":"HTTP/1.1","method":"GET","host":"my.actual.domain","uri":"/status.php","headers":{"X-Forwarded-For":["XX.XX.XX.XX"],"X-Forwarded-Proto":["https"],"X-Request-Id":["e7d70adf-f062-4318-b896-9dcc6427cc26"],"User-Agent":["Mozilla/5.0 (Windows) mirall/3.4.0stable-Win64 (build 20211129) (Nextcloud, windows-10.0.19044 ClientArchitecture: x86_64 OsArchitecture: x86_64)"],"Accept":["*/*"],"Authorization":[],"Cookie":[],"X-Forwarded-Host":["my.actual.domain"],"Accept-Encoding":["gzip, deflate"],"Accept-Language":["de-DE,en,*"]}},"duration":0.000655135,"status":502,"err_id":"swsppuht9","err_trace":"reverseproxy.statusError (reverseproxy.go:1196)"}

{"remote_ip":"172.17.0.1","remote_port":"48918","proto":"HTTP/1.1","method":"GET","host":"my.actual.domain","uri":"/","headers":{"User-Agent":["curl/7.68.0"],"Accept":["*/*"]}},"duration":0.000501788,"status":502,"err_id":"pr0bdzrzy","err_trace":"reverseproxy.statusError (reverseproxy.go:1196)"}
</code></details>

<details><summary>Changed Caddy config</summary>
<pre><code>
my.actual.domain {
    reverse_proxy localhost:11000
}
</code></pre>
</details>

**Comment by szaimen** [comment, marked as answer]
So it looks like apache inside the container is somehow not running. I don't know why and cannot reproduce this. It is probably something with docker itself that is not working correctly.

I know that it doesn't really help but I know of a guy that reinstalled its OS after having trouble with docker and afterwards it worked for him.

**Reply by tfedder-de** [reply]
It was indeed the apache (within the apache container) not working correctly. I had a look at the Caddyfile and it turned out the other routes were working.

I couldn't find out why it wasn't working. But after multiple reinstallation of the OS, I found out the hard way that the v-server I was using was not fully docker compatible (despite contradicting adversitement), probably due to the virtualization platform.

I switched to another server, where AIO is working fine so far. 👍 

Now I can finally try to migrate at least the user files from my dead raspi/ncpi.

Thanks for your help. 🙏
